### PR TITLE
Ts brush select

### PR DESCRIPTION
**Pull request by Yanyan-Wang**
- feat: automatically add draggable to keyShape, users do not need to assign it when custom a node or an edge; 
- feat: timing event for tree graph layout with beforelayout and afterlayout. closes: #1253.
- fix: cannot read 0 of null problem in getPointByCanvas; 
- fix: the shape.attr error in updateShapeStyle; 
- fix: modify the usage of bbox in view controller since the interface is changed by G; 
- fix: brush-select bug; 
- fix: update G version to solve lacking of removeChild problem;
- doc: update the demo of custom behavior doc;
- doc: add plugin demos and cases for site;
- doc: fix the shouldUpdate problem in treeWithLargeData demo on the site. closes: #1246;
- doc: changeData demo localRefresh. closes: #1234;
-  closes: #1250.

### CODE REVIEW

File: src/graph/graph.ts [hunk 1]
```diff
@@ -950,6 +951,10 @@ export default class Graph extends EventEmitter implements IGraph {
       return this;
     }
 
+    const canvas = this.get('canvas');
+    const localRefresh: boolean = canvas.get('localRefresh');
+    canvas.set('localRefresh', false);
+
```
Comment (baizn) [inline src/graph/graph.ts]: 是完全关闭局部渲染还是临时关闭?

File: src/item/item.ts [hunk 2]
```diff
@@ -138,6 +138,7 @@ export default class ItemBase implements IItemBase {
     if (keyShape) {
       keyShape.isKeyShape = true;
       self.set('keyShape', keyShape);
+      keyShape.set('draggable', true);
```
Comment (baizn) [inline src/item/item.ts]: 这里加了这个，自定义 node 里面的 draggable 就可以不加了。

File: src/graph/graph.ts (hunk 1 above)
Comment (Yanyan-Wang) [reply src/graph/graph.ts]: 只是在 changeData 的时候关一下，因为现在 G 的局部渲染还有问题，会导致更换数据的时候来不及完全清空，导致有上一步残留的图。

File: src/item/item.ts (hunk 2 above)
Comment (Yanyan-Wang) [reply src/item/item.ts]: 你是说 demo 里面的自定义节点吗？

**Pull request merged**